src/cache-manager.js

```diff
@@ -91,25 +110,46 @@ class CacheManager {
         addresses: JSON.stringify(inputUrlInfo.addresses),
       };
 
+      if (inputUrlInfo.meta.headers['content-length']) {
+        storageMetadata['upstream-content-length'] = inputUrlInfo.meta.headers['content-length'];
```

Comment:
this should only be used for debugging, and not deployed in production... Yeah, it does no harm, but it's really messy.